core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -218,6 +219,10 @@ public boolean isClosed() {
    */
   public void close() {
     if (!isClosed()) {
+      if (buffer != null && buffer.readableBytes() == 0) {
```

Comment:
I'd just assume not have this, since it is currently impossible. If you want to keep it, put a comment that it is for future safety or some such.